TypeCobol/Compiler/Scanner/Scanner.cs

```diff
@@ -959,11 +959,15 @@ public Token GetNextToken()
                 case ':':
                     // -- TypeCobol specific syntax --
                     // QualifiedNameSeparator => qualifierName::qualifiedName
-                    if (currentIndex < lastIndex && line[currentIndex + 1] == ':')
+                    if (currentIndex < lastIndex && line[currentIndex + 1] == ':' && (this.compilerOptions == null || !this.compilerOptions.AreForCopyParsing))
                     {
-                        // consume two :: chars
-                        currentIndex += 2;
-                        return new Token(TokenType.QualifiedNameSeparator, startIndex, startIndex + 1, tokensLine);
+                        // consume two :: chars in token1 and token2
+                        var token1 = ScanOneChar(startIndex, TokenType.ColonSeparator);
+                        var token2 = ScanOneChar(startIndex + 1, TokenType.ColonSeparator);
+                        GroupToken group = new GroupToken(TokenType.QualifiedNameSeparator, startIndex, startIndex + 1, tokensLine);
```

Comment:
Why use a GroupToken for token `::` now ?

Reply:
Group of Tokens is needed here in order to still returning one token when GetNextToken() is called, The REPLACE parsing will flat the group into it's groups tokens, and I just wanted to still returning TokenType.QualifiedNameSeparator.
But In fact maybe Group of Token can be avoid if we consider the flag  this.compilerOptions.AreForCopyParsing for a Pure Cobol Treatment not TypeCobol.

Reply:
After more investigation, characters :: that need to be replaced, must be generated as a partial Cobol Word Token.
E.g. :
`move W-Var1:: to W-Var2`
`W-Var1::` must be single a single Token of type Partial Cobol Word Token. Otherwise the replacing won't occurs.